### DISCUSSION TITLE
Base class in a submodule, derived class in some other submodule.

### DISCUSSION THREAD

**Discussion by andre-caldas** [wjakob/nanobind#538] [Q&A] [answered]
I have a module `A`. It has two submodules: `S` and `T`.

To make things more complicated, each module is defined in a different cpp translation unit:
```cpp
NB_MODULE(paracadis, m) {
  m.doc() = "ParaCADis python interface.";
  init_naming_scheme(m);
  init_geometric_primitives(m);
  init_document_tree(m);
}
```

Each method called above creates a submodule. Many classes in `geometric_primitives` derive from a class in `naming_scheme`. Because they are all "something with a name". :-)

Speaking in more abstract terms, in `A.S` I have a "base class":
```cpp
  nb::class_<BaseClass>(S, "BaseClass");
```

In `A.T`, I have a class derived from `BaseClass`:
```cpp
  // Does not seem to work.
  nb::class_<DerivedClass, BaseClass>(T, "DerivedClass");
  // Maybe, something like this... but where do I get `b` from?
  nb::class_<DerivedClass>(T, "DerivedClass", b);
```

What would be the correct way to setup bindings for `DerivedClass`?

**Comment by wjakob** [comment, marked as answer]
I think that should be supported. You can give a Python object to `nb::class_` to use as base, and you can use `nb::module_::import_("..").attr("..")` to fetch that object. Please refer to the documentation, this case is explained there somewhere.

**Reply by andre-caldas** [reply]
Dear professor,

Thank you very much for your help and patience. :-)

I thought you would like the feedback... so here is the implemented solution:
https://github.com/andre-caldas/ParaCADis/blob/016f10e3d572dd7105bf30103fdc5aa02e356d9a/src/base/geometric_primitives/python_bindings/module.cpp#L46
